xclim/sdba/adjustment.py

```diff
@@ -121,13 +125,47 @@ def _check_inputs(cls, *inputs, group):
             )
 
     @classmethod
-    def _harmonize_units(cls, *inputs, target: str | None = None):
+    def _harmonize_units(cls, *inputs, target: dict[str] | str | None = None):
         """Convert all inputs to the same units.
 
         If the target unit is not given, the units of the first input are used.
 
         Returns the converted inputs and the target units.
         """
+
+        def _harmonize_units_multivariate(
+            *inputs, dim, target: dict[str] | None = None
+        ):
+            def _convert_units_to(inda, dim, target):
+                for iv, v in enumerate(inda[dim].values):
+                    inda.attrs["units"], inda.attrs["standard_name"] = (
+                        inda[dim].attrs[a][iv] for a in ["_units", "_standard_name"]
+                    )
+                    inda[{dim: iv}] = convert_units_to(
+                        inda[{dim: iv}], target[v], context="infer"
+                    )
+                inda.attrs["units"], inda.attrs["standard_name"] = "", ""
+                return inda
+
+            if target is None:
+                target = {
+                    v: inputs[0][dim].attrs["_units"][iv]
+                    for iv, v in enumerate(inputs[0][dim].values)
+                }
+            return (
+                _convert_units_to(inda, dim=dim, target=target) for inda in inputs
+            ), target
```

Comment:
I tried putting `_harmonize_units_multivariate` as another class function so that , but I was getting error about generators. This is because of the structure of `inputs`. I can re-try if we judge it would be better like this. 

Also, we now an output type which is not certain, target returned can be a str or dict[str]. I know it's not ideal, but it was very convenient to write like this.